### ISSUE TITLE
[23] ECJ rejects a default case when it shouldn't

### ISSUE THREAD

**Issue by jarthana** [eclipse-jdt/eclipse.jdt.core#2933]
Here's the test:
```
public class X {
	public static float byteToFloat(Byte a) {
		return switch (a) {
			case float y -> y;
			default -> 2;
		};
	}
}
```
ECJ report on the default that Switch case cannot have both unconditional pattern and default label". Looks like somewhere we are omitting the fact that the float is not matching Byte.

**Comment by mpalat** [comment]
with the latest code the error had gone but the run-time results were wrong - it was branching to default- this gets corrected via the pr #2944

**Comment by stephan-herrmann** [comment]
Resolved by https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2944

I made a private note that we still  don't have a test case that systematically tests unbox-then-widen in pattern switches. To be added when at leisure.
